src/charm.py

```diff
@@ -95,6 +175,117 @@ def _pebble_layer(self) -> ops.pebble.LayerDict:
             },
         }
 
+    @property
+    def _application_bind_address(self) -> str | None:
+        binding = self.model.get_binding("juju-info")
+        if not binding:
+            return None
+        if not binding.network:
+            return None
+        if not binding.network.bind_address:
+            return None
+        return str(binding.network.bind_address)
+
+    @property
+    def _certificate_requests_table(self) -> List[dict[str, str]] | None:
+        # TODO: Error checking
+        r = requests.get(
+            url=f"https://{self._application_bind_address}:{self.port}/api/v1/certificate_requests",
+            verify=f"{CHARM_PATH}/{CONFIG_MOUNT}/0/ca.pem",
+        )
+        gocert_csrs = r.json()
+        return gocert_csrs
+
+    ## Status Checks ##
+    def _storages_attached(self) -> bool:
+        """Return if the storages are attached."""
+        return self.model.storages.get("config") and self.model.storages.get("database")  # type: ignore
+
+    def _gocert_available(self) -> bool:
+        """Return if the gocert server is reachable."""
+        try:
+            req = requests.get(
+                f"https://{self._application_bind_address}:{self.port}/status",
+                verify=f"{CHARM_PATH}/{CONFIG_MOUNT}/0/ca.pem",
+            )
+        except (requests.RequestException, OSError):
+            return False
+        if req.status_code != 200:
+            return False
+        return True
+
+    def _gocert_initialized(self) -> bool:
+        """Return if gocert is initialized."""
+        try:
+            req = requests.get(
+                f"https://{self._application_bind_address}:{self.port}/status",
+                verify=f"{CHARM_PATH}/{CONFIG_MOUNT}/0/ca.pem",
+            )
+        except (requests.RequestException, OSError):
```

Comment:
Why OSError?

Reply:
if the file in the path given in the verify argument does not exist, this call will raise an OSError. it gets resolved during configuration when we create and save the ca.pem file.